guis/common/db_classes/straw.py

```diff
@@ -145,6 +149,50 @@ def locate(self):
             .all()
         )
 
+    # same as above but returns ALL locations, not just the current one
+    # example: [<16499606623611227 straw position 84 on 1649960662278015 straw location LPAL0486 at LPALID01>, <16499477136822270 straw position 84 on 1649947713619747 straw location MN0238>]
+    def locations(self):
+        return (
+            DM.query(sl.StrawPosition)  # Get all the straw positions
+            # .join(sl.StrawPosition, sl.StrawPosition.location == sl.StrawLocation.id) # where straw positions have an entry in the straw location
+            .join(
+                sl.StrawPresent, sl.StrawPresent.position == sl.StrawPosition.id
+            )  # that also have straw present entries
+            .filter(sl.StrawPresent.straw == self.id)  # for this straw
+            .all()
+        )
+
+    # get the "present" occurance in straw_present
+    def currentPresence(self):
+        return (
+            DM.query(
+                        sl.StrawPresent.id,
+                        sl.StrawPresent.position,
+                        sl.StrawPresent.present,
+                        sl.StrawPresent.time_in,
+                        sl.StrawPresent.time_out
+                    )
+            .filter(sl.StrawPresent.straw == self.id)
+            .filter(sl.StrawPresent.present == 1)
+            .all()
+        )
+
+    # get all occurances in straw_position
+    def presences(self):
```

Comment:
Ah I see. OK so here's how this should work:

`def queryStrawPresentEntries` should return a query object.
`def getCurrentStrawPresent` should call the above function and filter it with `present == 1` and then call `one_or_none`.
And finally `getAllPastAndCurrentStrawPresentEntries` should call the same query function and then call `all` on the query.

Reply:
`DM.query(sl.StrawPresent).filter(.....)`

Reply:
We want to return the `StrawPresent` object(s) itself.